geode-core/src/main/java/com/gemstone/gemfire/admin/AlertLevel.java

```diff
@@ -145,8 +147,8 @@ public boolean equals(Object other) {
 		final AlertLevel that = (AlertLevel) other;
 
 		if (this.severity != that.severity) return false;
-		if (this.name != that.name &&
-	  		!(this.name != null &&
+		if (!Objects.equals(this.name, that.name) &&
+        !(this.name != null &&
 	  		this.name.equals(that.name))) return false;
 
```

Comment:
I think that the point of this code is to compare whether or not the 'other' name is the equivalent string to this.name.  A better way to test for that is to move the NPE check up, then compare:

		if (this.name != null && 
	  		this.name.equals(that.name)) return false;